allennlp/common/detectron.py

```diff
@@ -9,13 +9,36 @@ class DetectronConfig(NamedTuple):
     overrides: Optional[Dict[str, Any]] = None
 
 
-def apply_dict_to_cfg(d: Dict[str, Any], c: CfgNode) -> None:
-    for key, value in d.items():
-        key = key.upper()
-        if isinstance(value, dict):
-            apply_dict_to_cfg(value, c[key])
+# def apply_dict_to_cfg(d: Dict[str, Any], c: CfgNode) -> None:
```

Comment:
Why did you change this? The new code is also fine, no problem. I'm just wondering if I missed something when I wrote `apply_dict_to_cfg()`.

Reply:
There are some attributes (ex. cfg.MODEL.ROI_ATTRIBUTE_HEAD) that are not in the original detectron2 CfgNode. Accessing those keys with c[key] causes KeyError.

Reply:
Sure, but those new attributes could have been added with `applt_dict_to_cfg()`, right?

Reply:
You might be right, I just couldn't find a simple way how to handle nested CfgNode classes.
CfgNode sometimes has CfgNodes as an attribute (see https://github.com/allenai/allennlp/pull/4481/files#diff-b2b05f69be08bf21366facbbe9abfc5dR94).